src/django/api/views/facility/facilities_view_set.py

```diff
@@ -844,129 +839,84 @@ def claim(self, request, pk=None):
-                    workers_count = None
-                elif not validate_workers_count(workers_count):
-                    workers_count = None
-
-            except (ValueError, TypeError):
-                workers_count = None
-
-            facility_claim.facility_workers_count = workers_count
-
-            facility_claim.\
-                facility_name_native_language = local_language_name
+            sectors = validated_data.get("sectors")
+            if sectors:
+                facility_claim.sectors = sectors
+                if sectors:
+                    setattr(facility_claim, 'sector', sectors)
 
             facility_claim.save()
 
             for file in files:
-                timestamp = datetime.now().strftime('%Y%m%d%H%M%S')
-                milliseconds = datetime.now().strftime('%f')[:-3]
-                timestamp_ms = f'{timestamp}{milliseconds}'
-                file_name, file_extension = os.path.splitext(file.name)
-                file.name = (
-                    f'{slugify(file_name, allow_unicode=True)}-'
-                    f'{contributor.name}-{timestamp_ms}{file_extension}'
-                )
-                FacilityClaimAttachments.objects.create(
-                    claim=facility_claim,
-                    file_name=f'{slugify(file_name)}.{file_extension}',
-                    claim_attachment=file
+                self.__handle_file_upload(
+                    file,
+                    contributor.name,
+                    facility_claim
                 )
 
             send_claim_facility_confirmation_email(request, facility_claim)
             Facility.update_facility_updated_at_field(facility.id)
 
-            approved = (
-                FacilityClaim
-                .objects
-                .filter(status=FacilityClaimStatuses.APPROVED)
-                .filter(contributor=contributor)
-                .values_list('facility__id', flat=True)
-            )
-            pending = (
-                FacilityClaim
-                .objects
-                .filter(status=FacilityClaimStatuses.PENDING)
-                .filter(contributor=contributor)
-                .values_list('facility__id', flat=True)
-            )
+            approved = FacilityClaim.objects.filter(
+                status=FacilityClaimStatuses.APPROVED,
+                contributor=contributor
+            ).values_list('facility__id', flat=True)
+
+            pending = FacilityClaim.objects.filter(
+                status=FacilityClaimStatuses.PENDING,
+                contributor=contributor
+            ).values_list('facility__id', flat=True)
+
             return Response({
                 'pending': pending,
                 'approved': approved,
             })
-        except Facility.DoesNotExist as exc:
-            raise NotFound() from exc
-        except Contributor.DoesNotExist as exc:
+
+        except (Facility.DoesNotExist, Contributor.DoesNotExist) as exc:
```

Comment:
Please separate the exception of these errors as it was before changes, and add a `detail` message for each one:  

```python

except Facility.DoesNotExist as exc:
            raise NotFound(detail='Facility not found.') from exc
except Contributor.DoesNotExist as exc:
            raise NotFound(detail='Contributor not found.') from exc
```

Reply:
Done, thank you.

Reply:
Upd: I introduced DRFValidationError regarding this suggestion: https://github.com/opensupplyhub/open-supply-hub/pull/578#pullrequestreview-2759141143